File: mobile/lib/models/video_detail_info.dart
/// 视频详情信息
class VideoDetailInfo {
  /// 视频 BV 号
  final String bvid;

  /// 视频 AV 号
  final int aid;

  /// 第一个分 P 的 cid
  final int cid;

  /// 视频标题
  final String title;

  /// 视频简介
  final String desc;

  /// 视频封面
  final String cover;

  /// UP 主信息
  final OwnerInfo owner;

  /// 视频数据统计
  final VideoStat stat;

  /// 发布时间戳
  final int pubdate;

  /// 视频时长（秒）
  final int duration;

  /// 分 P 数量
  final int videos;

  const VideoDetailInfo({
    required this.bvid,
    required this.aid,
    required this.cid,
    required this.title,
    required this.desc,
    required this.cover,
    required this.owner,
    required this.stat,
    required this.pubdate,
    required this.duration,
    required this.videos,
  });

  factory VideoDetailInfo.fromJson(Map<String, dynamic> json) {
    var cover = json['pic'] as String? ?? '';
    if (cover.startsWith('//')) {
      cover = 'https:$cover';
    }

    return VideoDetailInfo(
      bvid: json['bvid'] as String? ?? '',
      aid: json['aid'] as int? ?? 0,
      cid: json['cid'] as int? ?? 0,
      title: json['title'] as String? ?? '',
      desc: json['desc'] as String? ?? '',
      cover: cover,
      owner: OwnerInfo.fromJson(json['owner'] as Map<String, dynamic>? ?? {}),
      stat: VideoStat.fromJson(json['stat'] as Map<String, dynamic>? ?? {}),
      pubdate: json['pubdate'] as int? ?? 0,
      duration: json['duration'] as int? ?? 0,
      videos: json['videos'] as int? ?? 1,
    );
  }

  /// 格式化发布时间
  String get formattedPubdate {
    final date = DateTime.fromMillisecondsSinceEpoch(pubdate * 1000);
    return '${date.year}-${date.month.toString().padLeft(2, '0')}-${date.day.toString().padLeft(2, '0')}';
  }

  /// 格式化时长
  String get formattedDuration {
    final hours = duration ~/ 3600;
    final minutes = (duration % 3600) ~/ 60;
    final seconds = duration % 60;

    if (hours > 0) {
      return '$hours:${minutes.toString().padLeft(2, '0')}:${seconds.toString().padLeft(2, '0')}';
    }
    return '${minutes.toString().padLeft(2, '0')}:${seconds.toString().padLeft(2, '0')}';
  }
}

/// UP 主信息
class OwnerInfo {
  final int mid;
  final String name;
  final String face;

  const OwnerInfo({required this.mid, required this.name, required this.face});

  factory OwnerInfo.fromJson(Map<String, dynamic> json) {
    var face = json['face'] as String? ?? '';
    if (face.startsWith('//')) {
      face = 'https:$face';
    }

    return OwnerInfo(
      mid: json['mid'] as int? ?? 0,
      name: json['name'] as String? ?? '',
      face: face,
    );
  }
}

/// 视频统计数据
class VideoStat {
  final int view;
  final int danmaku;
  final int reply;
  final int favorite;
  final int coin;
  final int share;
  final int like;

  const VideoStat({
    required this.view,
    required this.danmaku,
    required this.reply,
    required this.favorite,
    required this.coin,
    required this.share,
    required this.like,
  });

  factory VideoStat.fromJson(Map<String, dynamic> json) {
    return VideoStat(
      view: json['view'] as int? ?? 0,
      danmaku: json['danmaku'] as int? ?? 0,
      reply: json['reply'] as int? ?? 0,
      favorite: json['favorite'] as int? ?? 0,
      coin: json['coin'] as int? ?? 0,
      share: json['share'] as int? ?? 0,
      like: json['like'] as int? ?? 0,
    );
  }

  /// 格式化播放量
  String get formattedView {
    if (view >= 100000000) {
      return '${(view / 100000000).toStringAsFixed(1)}亿';
    } else if (view >= 10000) {
      return '${(view / 10000).toStringAsFixed(1)}万';
    }
    return view.toString();
  }

  /// 格式化弹幕数
  String get formattedDanmaku {
    if (danmaku >= 10000) {
      return '${(danmaku / 10000).toStringAsFixed(1)}万';
    }
    return danmaku.toString();
  }
}

File: windows/Models/VideoDetailInfo.cs
using System;
using System.Globalization;
using System.Text.Json;

namespace bilibili_music_player_windows.Models
{
    /// <summary>
    /// 视频详情信息，从 /x/web-interface/view API 返回。
    /// </summary>
    public sealed class VideoDetailInfo
    {
        /// <summary>视频 BV 号。</summary>
        public string Bvid { get; init; } = string.Empty;

        /// <summary>视频 AV 号。</summary>
        public long Aid { get; init; }

        /// <summary>第一个分 P 的 cid。</summary>
        public long Cid { get; init; }

        /// <summary>视频标题。</summary>
        public string Title { get; init; } = string.Empty;

        /// <summary>视频简介。</summary>
        public string Desc { get; init; } = string.Empty;

        /// <summary>视频封面 URL。</summary>
        public string Cover { get; init; } = string.Empty;

        /// <summary>UP 主信息。</summary>
        public OwnerInfo Owner { get; init; } = new();

        /// <summary>视频数据统计。</summary>
        public VideoStat Stat { get; init; } = new();

        /// <summary>发布时间戳（Unix 秒）。</summary>
        public long Pubdate { get; init; }

        /// <summary>视频时长（秒）。</summary>
        public long Duration { get; init; }

        /// <summary>分 P 数量。</summary>
        public long Videos { get; init; } = 1;

        /// <summary>格式化发布时间。</summary>
        public string FormattedPubdate
        {
            get
            {
                if (Pubdate <= 0)
                {
                    return string.Empty;
                }

                var date = DateTimeOffset.FromUnixTimeSeconds(Pubdate).LocalDateTime;
                return date.ToString("yyyy-MM-dd", CultureInfo.InvariantCulture);
            }
        }

        /// <summary>格式化时长。</summary>
        public string FormattedDuration
        {
            get
            {
                var hours = Duration / 3600;
                var minutes = (Duration % 3600) / 60;
                var seconds = Duration % 60;

                if (hours > 0)
                {
                    return $"{hours}:{minutes:D2}:{seconds:D2}";
                }

                return $"{minutes:D2}:{seconds:D2}";
            }
        }

        public static VideoDetailInfo FromJsonElement(JsonElement dataElement)
        {
            var cover = GetNullableString(dataElement, "pic") ?? string.Empty;
            if (cover.StartsWith("//", StringComparison.Ordinal))
            {
                cover = $"https:{cover}";
            }

            OwnerInfo owner;
            if (dataElement.TryGetProperty("owner", out var ownerElement))
            {
                owner = OwnerInfo.FromJsonElement(ownerElement);
            }
            else
            {
                owner = new OwnerInfo();
            }

            VideoStat stat;
            if (dataElement.TryGetProperty("stat", out var statElement))
            {
                stat = VideoStat.FromJsonElement(statElement);
            }
            else
            {
                stat = new VideoStat();
            }

            return new VideoDetailInfo
            {
                Bvid = GetNullableString(dataElement, "bvid") ?? string.Empty,
                Aid = ReadInt64(dataElement, "aid") ?? 0,
                Cid = ReadInt64(dataElement, "cid") ?? 0,
                Title = GetNullableString(dataElement, "title") ?? string.Empty,
                Desc = GetNullableString(dataElement, "desc") ?? string.Empty,
                Cover = cover,
                Owner = owner,
                Stat = stat,
                Pubdate = ReadInt64(dataElement, "pubdate") ?? 0,
                Duration = ReadInt64(dataElement, "duration") ?? 0,
                Videos = ReadInt64(dataElement, "videos") ?? 1,
            };
        }

        private static string? GetNullableString(JsonElement element, string propertyName)
        {
            if (!element.TryGetProperty(propertyName, out var value))
            {
                return null;
            }

            return value.ValueKind switch
            {
                JsonValueKind.String => value.GetString(),
                JsonValueKind.Number => value.GetRawText(),
                _ => value.GetRawText(),
            };
        }

        private static long? ReadInt64(JsonElement element, string propertyName)
        {
            if (!element.TryGetProperty(propertyName, out var value))
            {
                return null;
            }

            if (value.ValueKind == JsonValueKind.Number && value.TryGetInt64(out var number))
            {
                return number;
            }

            if (value.ValueKind == JsonValueKind.String &&
                long.TryParse(value.GetString(), NumberStyles.Integer,
                    CultureInfo.InvariantCulture, out var parsed))
            {
                return parsed;
            }

            return null;
        }
    }
}
